### ISSUE TITLE
Switch over to JS typed arrays (Uint8Array, Uint32Array)

### ISSUE THREAD

**Issue by erkyrath** [erkyrath/quixe#25]
The last time I tested this, it wasn't any faster than regular JS arrays. However, that was many browser optimizations ago; it might be a good idea now.

We have seen problems with very large Blorb files (hundreds of megabytes), which can be represented in-memory as Node Buffers or Uint8Arrays, but not naive Number arrays.

**Comment by curiousdannii** [comment]
Probably easiest to use DataViews. I [made an extension to DataView](https://github.com/curiousdannii/ifvms.js/blob/master/src/common/utils.js#L50) which also has support for getting and setting a block of data too.

**Comment by erkyrath** [comment]
Well, that was only hanging around for eight years, right?

Done now.

The speedup remains tiny (maybe 3%). I suspect JS engine optimizations have long since gotten to the point where they do this automatically. However, it's tidier to do it explicitly.

**Comment by curiousdannii** [comment]
Does it support a Uint8Array being provided directly?

GlkOte would also need to support typed arrays for the full benefits to be apparent.

**Comment by erkyrath** [comment]
You can pass in a Uint8Array or an Array by writing

```
GiLoad.load_run(OPTIONS, array, { format:'array' })
```

This covers Quixe changes. GlkApi hasn't broken as far as I can tell, and I ran all the usual unit tests. (Well, not _all_ of them, but the ones dealing with Unicode and Blorb image loading.)

**Comment by erkyrath** [comment]
Nope, nothing broken in the unit tests. I'm not using Uint32Array anywhere, so there's no reason to have triggered the error you discovered there.

**Comment by erkyrath** [comment]
I still want to try using Uint32Array for the stack, or (more importantly) frame.valstack.

**Comment by curiousdannii** [comment]
I found for Z-Machine that it was really nice to store the stack in the Quetzal format, except for needing to swap the bytes before saving. It might be similar for Glulx. There are JS libraries for swapping bytes, but I haven't looked into them specifically yet.
